File: Basics_with_Python/PythonBasicCourse/5_While Loop/12_While Loop - Exercise/More_Exercises/02. Report System.py
funds_required = int(input())
funds_collected = 0
counter = 1
cash_sum = 0
card_sum = 0
cash_counter = 0
card_counter = 0
bill = 0
while funds_collected < funds_required:
    bill = input()
    if bill == "End":
        break
    bill = int(bill)
    if counter % 2 != 0:  # cash only
        if bill <= 100:
            print("Product sold!")
            funds_collected += bill
            cash_sum += bill
            counter += 1
            cash_counter += 1
        else:
            print("Error in transaction!")
            counter += 1
    else:  # card only
        if bill >= 10:
            print("Product sold!")
            funds_collected += bill
            card_sum += bill
            counter += 1
            card_counter += 1
        else:
            print("Error in transaction!")
            counter += 1
if bill == "End":
    # print("Error tests:  funds collected = ",funds_collected)
    print("Failed to collect required money for charity.")
else:
    average_cash = cash_sum / cash_counter
    average_card = card_sum / card_counter
    print(f"Average CS: {average_cash:.2f}")
    print(f"Average CC: {average_card:.2f}")

File: Basics_with_JavaScript/1112 While Loop/More Exersises/02. Report System.js
function vladi(array) {
  let cashSum = 0;
  let cardSum = 0;
  let cashCount = 0;
  let cardCount = 0;
  let expectedIncome = Number(array[0]);
  let income = 0;
  let isMoneyGathered = false;

  let index = 0;
  index++;
  let command = array[index];

  while (command !== "End") {
    let currentItemPrice = Number(command);

    if (index % 2 !== 0 && currentItemPrice > 100) {
      console.log(`Error in transaction!`);
    } else if (index % 2 !== 0 && currentItemPrice <= 100) {
      income += currentItemPrice;
      cashSum += currentItemPrice;
      cashCount += 1;
      console.log(`Product sold!`);
    } else if (index % 2 == 0 && currentItemPrice < 10) {
      console.log(`Error in transaction!`);
    } else if (index % 2 == 0 && currentItemPrice >= 10) {
      income += currentItemPrice;
      cardSum += currentItemPrice;
      cardCount += 1;
      console.log(`Product sold!`);
    }

    if (expectedIncome <= income) {
      isMoneyGathered = true;
      break;
    }

    index++;
    command = array[index];
    
  }

  let averageCash = (cashSum / cashCount).toFixed(2);
  let averageCard = (cardSum / cardCount).toFixed(2);

  if (isMoneyGathered) {
    console.log(`Average CS: ${averageCash}`);
    console.log(`Average CC: ${averageCard}`);
  } else {
    console.log(`Failed to collect required money for charity.`);
 }
} 

vladi(["500", "120", "8", "63", "256", "78", "317"]);

vladi(["600",
"86",
"150",
"98",
"227",
"End"]);
